### PR TITLE
Adds missing unit tests for LayerBuildDefinition in build_graph

### PR DESCRIPTION
**Pull request by asgarciap**
#### Which issue(s) does this change fix?

NONE

#### Why is this change necessary?
There are no unit tests for LayerBuildDefinition in the build_graph unit test file, and eventhough its behaviour and implementation is quite similar to FunctionBuildDefinition, since they are separate clases and lists, we should also have it covered in the unit testing workflow.

#### How does it address the issue?
It adds some missing asserts and a new test_case for LayerBuildDefinition.

#### What side effects does this change have?
None I am aware of.

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by sriram-mv — APPROVED** [review]

**Review by aahung — APPROVED** [review]

**Comment by asgarciap** [comment]
Hi guys, this PR have been approved, would be nice if anyone with enough privs can merge it or let me know if there is anything else I should do before it can be merged. Thanks. 😁

**Comment by aahung** [comment]
merging soon, thx for reminding us

**Pull request merged**
